src/test/transaction_tests.cpp

```diff
@@ -370,6 +369,40 @@ BOOST_AUTO_TEST_CASE(tx_invalid)
     }
 }
 
+BOOST_AUTO_TEST_CASE(tx_no_inputs)
+{
+    CMutableTransaction empty;
+
+    TxValidationState state;
+    BOOST_CHECK_MESSAGE(!CheckTransaction(CTransaction(empty), state), "Transaction with no inputs should be invalid.");
+    BOOST_CHECK(state.GetRejectReason() == "bad-txns-vin-empty");
+}
+
+BOOST_AUTO_TEST_CASE(tx_oversized)
+{
+    auto maxPayloadSize = MAX_BLOCK_WEIGHT / WITNESS_SCALE_FACTOR - 70;
+    {
+        CMutableTransaction tx;
+        tx.vin.resize(1);
+        tx.vout.emplace_back(1, CScript() << OP_RETURN << std::vector<unsigned char>(maxPayloadSize));
+
+        TxValidationState state;
+        CheckTransaction(CTransaction(tx), state);
```

Comment:
Am I missing something simple?  In https://github.com/bitcoin/bitcoin/pull/29862#discussion_r1643298389, it was mentioned that -70 should pass, but -69 should not.  The test case appears to check that -70 fails with `bad-txns-oversize`.  If -70 passes the size checking, it would fail for a reason other than `bad-txns-oversize`, which might be worth checking (i.e. shows that CheckTransaction is proceeding past the `bad-txns-oversize` check).

Disclaimer:  I didn't check the txn byte math yet for 69/70 yet (e.g. version, inputcount, inputs,...), but the concept is the same.  Checking below and above the threshold makes sense.

Reply:
I would need to construct fully valid transaction for it to pass - since this doesn't have valid inputs it's considered a coinbase transaction which would need some extra data, which would be outside the scope of this validation. Since I'm only changing the outputs size and that triggers the correct error, I consider that to be enough.